server/server.go

```diff
@@ -229,9 +231,14 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 	projectLocker := &events.DefaultProjectLocker{
 		Locker: lockingClient,
 	}
+	parsedURL, err := ParseAtlantisURL(userConfig.AtlantisURL)
```

Comment:
I moved the call to `url.Parse` inside this function so that all the validation happened in the same place